releases/release-4.0.9.md

```diff
@@ -0,0 +1,127 @@
+    - Fix up displaying executor runtime info & Add runtime information for DML in explain analyze [#21066](https://github.com/pingcap/tidb/pull/21066)
+    - Deprecate `enable-streaming`. [#21055](https://github.com/pingcap/tidb/pull/21055)
+    - Avoid unnecessary warnings/errors when folding constants in shortcut-able expressions [#21040](https://github.com/pingcap/tidb/pull/21040)
+    - TiDB now provides an error for statements that use the syntax `LOCK IN SHARE MODE`. This helps alert users that this feature is not implemented by TiDB, but a workaround to restore the previous behavior is available by setting `tidb_enable_noop_functions=1`. [#21005](https://github.com/pingcap/tidb/pull/21005)
+    - Set correct default value for new added enum column. [#20998](https://github.com/pingcap/tidb/pull/20998)
+
++ TiKV
+
+    - Enable dynamically change config pessimistic-txn.pipelined. [#9100](https://github.com/tikv/tikv/pull/9100)
+    - Running checksum in BR and Lightning should have less influence on the cluster performance. [#9098](https://github.com/tikv/tikv/pull/9098)
+    - Add a metrics for ingest sst errors [#9096](https://github.com/tikv/tikv/pull/9096)
+    - Prevent hibernation when there are still peers catching up logs [#9093](https://github.com/tikv/tikv/pull/9093)
+    - Increase the success rate of pipelined pessimistic locking. [#9086](https://github.com/tikv/tikv/pull/9086)
+    - Add check to avoid corruption due to RocksDB block cache error #8243. Also disable `force-consistency-checks` by default to improve performance. [#9029](https://github.com/tikv/tikv/pull/9029)
+    - Change the default value of `apply-max-batch-size` and `store-max-batch-size` to 1024 [#9020](https://github.com/tikv/tikv/pull/9020)
+    - Add config for max-background-flushes [#8947](https://github.com/tikv/tikv/pull/8947)
+    - Enable the unified read pool for the storage module by default. [#8887](https://github.com/tikv/tikv/pull/8887)
+
++ PD
+
+    - Dashboard: update to v2020.11.26.1 [#3219](https://github.com/pingcap/pd/pull/3219)
+    - Check cluster version on stores become tombstone [#3213](https://github.com/pingcap/pd/pull/3213)
+    - The store of lower version is no longer allowed to change from `Tombstone` back to `Up`. [#3206](https://github.com/pingcap/pd/pull/3206)
+    - Fix a bug that service safe points with  infinite TTL may disappear. [#3143](https://github.com/pingcap/pd/pull/3143)
+
++ Tools
+
+    - TiCDC
+
+        * Metrics: add alert for tikv hibernate regions [#1120](https://github.com/pingcap/ticdc/pull/1120)
+        * Fix the bug that multiple owners could exist when owner campaign key is deleted [#1104](https://github.com/pingcap/ticdc/pull/1104)
+        * Kafka_consumer: fix wrong timezone when 'TZ' env is not set and downstream is mysql [#1095](https://github.com/pingcap/ticdc/pull/1095)
```

Comment:
```suggestion
        * Kafka_consumer: fix wrong timezone when 'TZ' env is not set and downstream is mysql [#1095](https://github.com/pingcap/ticdc/pull/1095)
        * Reduce memory usage in schema storage by avoiding table info replication [#1127](https://github.com/pingcap/ticdc/pull/1127)

# TiDB-binlog

## Improvements

* Exit drainer process with non-zero code if upstream PD is down or apply DDL/DML to downstream failed [#1012](https://github.com/pingcap/tidb-binlog/pull/1012)

# BR

## Improvements

* BR no longer accepts --checksum false in command line, which did not disable checksum. The correct usage always include the = sign: --checksum=false [#588](https://github.com/pingcap/br/pull/588)
* Support change PD config temporary [#596](https://github.com/pingcap/br/pull/596)
* Support analyze table after restore [#622](https://github.com/pingcap/br/pull/622)
* Retry for read index not ready and proposal in merging mode [#626](https://github.com/pingcap/br/pull/626)

## Bug Fixes

* Fix special characters in S3 secret access keys [#617](https://github.com/pingcap/br/pull/617)

# Dumpling

## Improvements

* Retry dumping on failed chunks [#182](https://github.com/pingcap/dumpling/pull/182)
* Support configuring both -F and -r arguments at the same time [#177](https://github.com/pingcap/dumpling/pull/177)
* Exclude system databases in --filter parameter by default [#194](https://github.com/pingcap/dumpling/pull/194)
* Support --transactional-consistency parameter and support rebuild mysql connections during retry [#199](https://github.com/pingcap/dumpling/pull/199)

## Bug Fixes

* fix the problem that dumpling may get blocked when its connection to database server is closed [#190](https://github.com/pingcap/dumpling/pull/190)

# TiDB-Lightning

## Improvements

* Filter out all system schemas by default [#459](https://github.com/pingcap/tidb-lightning/pull/459)
* Support set a default value for auto random primary key for local/importer backend [#457](https://github.com/pingcap/tidb-lightning/pull/457)
* Use range properties to make range split more precise with local backend [#422](https://github.com/pingcap/tidb-lightning/pull/422)
* `tikv-importer.region-split-size`, `mydumper.read-block-size`, `mydumper.batch-size` and `mydumper.max-region-size` can now accept human-readable format in the form "2.5 GiB" [#471](https://github.com/pingcap/tidb-lightning/pull/471)

## Bug Fixes

* Fix a bug about encoding data with wrong field info [#437](https://github.com/pingcap/tidb-lightning/pull/437)
* Fix a bug that gc life time ttl does not take effect [#448](https://github.com/pingcap/tidb-lightning/pull/448)
* Fix a bug that causes panic when manually stops Lightning import in localbackend mode [#484](https://github.com/pingcap/tidb-lightning/pull/484)

```

Reply:
Addd in https://github.com/pingcap/docs/pull/4294/commits/68de6db8c7e09deeaae3f6a7db45b60ed16f0ef7.